packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_global_client.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { UiSettingsClientCommon } from './ui_settings_client_common';
+import { UiSettingsServiceOptions } from '../types';
+import { BaseUiSettingsClient } from './base_ui_settings_client';
+import { SettingNotRegisteredError } from './not_registered_error';
+
+interface UserProvidedValue<T = unknown> {
+  userValue?: T;
+  isOverridden?: boolean;
+}
+
+type UserProvided<T = unknown> = Record<string, UserProvidedValue<T>>;
+
+/**
+ * Global UiSettingsClient
+ */
+export class UiSettingsGlobalClient extends BaseUiSettingsClient {
+  private readonly uiSettingsClientCommon;
+
+  constructor(options: UiSettingsServiceOptions) {
+    const { log, defaults = {}, overrides = {} } = options;
+    super({ overrides, defaults, log });
+    this.uiSettingsClientCommon = new UiSettingsClientCommon(options);
+  }
+
+  async getUserProvided<T = unknown>(): Promise<UserProvided<T>> {
+    return this.uiSettingsClientCommon.getUserProvided();
+  }
+
+  async setMany(changes: Record<string, any>) {
+    const registeredSettings = super.getRegistered();
+    Object.keys(changes).forEach((key) => {
+      if (!registeredSettings[key]) {
```

Comment:
I am not sure about the desired logic here. Do we wish to throw if *any one* of the settings is not registered and persist none of them? Or do we persist the ones that are registered, and throw for the others?

Reply:
To answer the question: probably throws for all if at least one is unknown. 

Now I have to ask another question 😄: why this difference of behavior in `set` and `setMany` between the 'normal' client and the global client?

Reply:
"the normal" one doesn't throw if the setting has not been registered. this is the thing we discussed here:
https://github.com/elastic/kibana/issues/144909#issuecomment-1322522616